aiosmtpd/smtp.py

```diff
@@ -535,14 +543,9 @@ def smtp_DATA(self, arg):
         received_data = EMPTYBYTES.join(data)
         if self._decode_data:
             received_data = received_data.decode('utf-8')
-        args = (self.peer, self.mailfrom, self.rcpttos, received_data)
-        kwargs = {}
-        if not self._decode_data:
-            kwargs = {
-                'mail_options': self.mail_options,
-                'rcpt_options': self.rcpt_options,
-                }
-        kwargs.update({'loop': self.loop})
+        self.message.received_data = received_data
+        args = (self.session, self.message)
+        kwargs = {'loop': self.loop}
```

Comment:
I dont think its abstraction breakage. Its aiosmtpd, it will be little bit strange if asyncio staff brakes something here.
Anyway, it we want to pass loop as param to handlers not as kwarg, session is best place for it. Moved it there. It can be easily changed in future, as it wouldn`t require to refactor all handlers with its removal.